verification/tagTCI.go

```diff
@@ -3,55 +3,239 @@
+	}
+}
+
+// Check whether the ExtendTCI command updates the current TCI and cumulative TCI.
+func TestExtendTCI(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var err error
+	useSimulation := false // To indicate that simulation context is not used
+
+	// Get default context handle
+	handle := getInitialContextHandle(d, c, t, useSimulation)
+
+	// Get digest size
+	profile, err := GetTransportProfile(d)
+	if err != nil {
+		t.Fatalf("Could not get profile: %v", err)
+	}
+	digestLen := profile.GetDigestSize()
 
-	if *handle != *ctx {
-		t.Errorf("New context handle from TagTCI was %x, expected %x", handle, ctx)
+	// Initialize TCI inputs with all zeroes
+	// since, TCI_DEFAULT for default context is all zeroes
+	defaultTci := make([]byte, digestLen)
+
+	tciValue := make([]byte, digestLen)
+	for i := range tciValue {
+		tciValue[i] = byte(i)
+	}
+
+	// Initialize hasher
+	var hasher hash.Hash
+	if digestLen == 32 {
+		hasher = sha256.New()
+	} else if digestLen == 48 {
+		hasher = sha512.New384()
+	} else {
+		t.Fatal("[FATAL]: Unsupported hash algorithm used for TCI value generation")
 	}
 
-	_, err = client.GetTaggedTCI(tag)
+	// Set current TCI value
+	_, err = c.ExtendTCI(handle, tciValue)
+	if err != nil {
+		t.Fatalf("Could not extend TCI: %v", err)
+	}
+
+	taggedTCI, err := c.GetTaggedTCI(defaultCtxTCITag)
 	if err != nil {
 		t.Fatalf("Could not get tagged TCI: %v", err)
 	}
 
-	// TODO: For profiles which use auto-initialization, we don't know the expected
-	// TCIs. Uncomment this once the DeriveChild API is implemented so the test
-	// can control the TCI inputs.
-	/*
-		wantCumulativeTCI := make([]byte, profile.GetDigestSize())
-		if !reflect.DeepEqual(taggedTCI.CumulativeTCI, wantCumulativeTCI) {
-			t.Errorf("GetTaggedTCI returned cumulative TCI %x, expected %x", taggedTCI.CumulativeTCI, wantCumulativeTCI)
-		}
+	// Check TCI_CURRENT
+	wantCurrentTCI := tciValue
+	if !bytes.Equal(taggedTCI.CurrentTCI, tciValue) {
+		t.Errorf("GetTaggedTCI returned current TCI %x, expected %x", taggedTCI.CurrentTCI, wantCurrentTCI)
+	}
 
-		wantCurrentTCI := make([]byte, profile.GetDigestSize())
-		if !reflect.DeepEqual(taggedTCI.CurrentTCI, wantCurrentTCI) {
-			t.Errorf("GetTaggedTCI returned current TCI %x, expected %x", taggedTCI.CurrentTCI, wantCurrentTCI)
-		}
-	*/
+	// Cross-verify TCI_CUMULATIVE
+	hasher.Write(defaultTci)
+	hasher.Write(taggedTCI.CurrentTCI)
+	if wantCumulativeTCI := hasher.Sum(nil); !bytes.Equal(taggedTCI.CumulativeTCI, wantCumulativeTCI) {
+		t.Errorf("GetTaggedTCI returned cumulative TCI %x, expected %x", taggedTCI.CumulativeTCI, wantCumulativeTCI)
+	}
+}
 
-	// Make sure some other tag is still not found.
-	if _, err := client.GetTaggedTCI(TCITag(98765)); !errors.Is(err, StatusBadTag) {
-		t.Fatalf("GetTaggedTCI returned %v, want %v", err, StatusBadTag)
+// Check whether the ExtendTCI command with derived child context.
+func TestExtendTciOnDerivedContexts(d TestDPEInstance, c DPEClient, t *testing.T) {
```

Comment:
See https://github.com/chipsalliance/caliptra-dpe/issues/257. I would recommend moving these tests to be ExtendTci test and use CertifyKey to get the TCIs (instead of TagTCI).

Reply:
I have added a new function in certifyKey.go that parses MultiuTcbInfo, gets cumulative/current values from FWID array and cross checks with hashed values.
Also, the new tests in TagTCI.go are moved to ExtendTci.go.
Please let me know the comments!